src/System.IO.Pipelines/src/System/IO/Pipelines/BufferSegment.cs

```diff
@@ -78,6 +78,31 @@ public void ResetMemory()
 
         public Memory<byte> AvailableMemory { get; private set; }
 
+        public IMemoryList<byte> GetNext(long offset, out int localIndex)
+        {
+            BufferSegment current = this;
```

Comment:
Should `this` BufferSegment class be marked as sealed just like the one in System.Memory?